services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/AwsConnectionServiceImpl.java

```diff
@@ -26,17 +25,17 @@ public class AwsConnectionServiceImpl implements AwsConnectionService {
     private AwsConnectionRepo awsConnectionRepo;
 
     @Override
-    public AwsDatasourceResponse saveConnectionInfo(AwsDatasourceRequest awsDatasourceRequest, Projects project) throws InvalidConnectionTypeException {
+    public AwsDatasourceResponse saveConnectionInfo(AwsDatasourceRequest awsDatasourceRequest, String userName) throws InvalidConnectionTypeException {
 
-        Optional<AWSDatasource> dataSources = awsConnectionRepo.findByDataSourceName(awsDatasourceRequest.getDataSourceName());
+        Optional<AWSDatasource> dataSources = awsConnectionRepo.findByDataSourceNameIgnoreCaseAndActive(awsDatasourceRequest.getDataSourceName().trim(),true);
         if (dataSources.isPresent()) {
-            throw new BadRequestException("Data source name already exist.");
+            throw new BadRequestException("Entered Data Source Name is already used.Please enter a different name");
         }
 
-        String connectionType = awsDatasourceRequest.getConnectionType();//access Type
+        String connectionType = awsDatasourceRequest.getConnectionType();
 
         if (IAM.getAccessType().equals(connectionType)) {
```

Comment:
Check whether enum has the passed connection type

Reply:
Added this condition